degica/komoju-woocommerce
# KOMOJU Fields integration

This PR (optionally) turns this plugin into a "direct" plugin that shows payment fields directly on the WooCommerce checkout page.

![image](https://user-images.githubusercontent.com/2793160/210722177-97d3fd11-1b84-477f-9feb-1473fb94e26d.png)

![image](https://user-images.githubusercontent.com/2793160/210722223-741c09fd-c4ce-4373-ad46-a2ae4009b6fd.png)

This setup utilizes our tokens API, so that sensitive information like credit card numbers do not ever touch the WooCommerce back-end.

Redirects to pages hosted by us may still occur in some cases, like 3DS and konbini payment instructions. We may further streamline UX in the future to avoid these redirects.

## How to test

1. `docker-compose down --volumes; docker-compose up` to make sure you're on fresh data
2. `(cd tests && npx cypress run)` to install WooCommerce and make sure everything works
3. Visit http://localhost:8000/?post_type=product
4. Add a product to cart then checkout
5. Fill in a fake address
6. You should see KOMOJU payment options. Credit Card and Konbini should show inline fields (if not, then go to /wp-admin, then woocommerce settings. make sure komoju is set up and payment methods active)
7. Paying using those methods should work (minus webhooks since it's localhost. can test webhooks with ngrok)
